File: src/main/java/com/team573/gongguri/domain/chat/entity/ChatRoom.java
package com.team573.gongguri.domain.chat.entity;

import com.team573.gongguri.global.entity.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "chat_room")
public class ChatRoom extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long chatRoomId;
}

File: src/main/java/com/team573/gongguri/domain/chat/entity/ChatRoom.kt
package com.team573.gongguri.domain.chat.entity

import com.team573.gongguri.global.entity.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "chat_room")
class ChatRoom(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val chatRoomId: Long? = null
) : BaseEntity() {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ChatRoom

        if (chatRoomId != other.chatRoomId) return false

        return true
    }

    override fun hashCode(): Int {
        return chatRoomId?.hashCode() ?: 0
    }
}
